go/vt/vttablet/tabletserver/vstreamer/uvstreamer.go

```diff
@@ -351,12 +366,18 @@ func (uvs *uvstreamer) init() error {
 		if err := uvs.setStreamStartPosition(); err != nil {
 			return err
 		}
-	} else if uvs.startPos == "" || len(uvs.inTablePKs) > 0 {
+	}
+
+	if uvs.startPos == "" || len(uvs.inTablePKs) > 0 {
 		if err := uvs.buildTablePlan(); err != nil {
 			return err
 		}
 	}
 
+	if uvs.startPos != "" && len(uvs.inTablePKs) > 0 {
+		uvs.pickingUpInputOffset = true
```

Comment:
`pickingUpInputOffset` acts like a new custom behavior type.
It depends on only the input parameters.

We aim that the existing logic is clearly intact if this type is false.

Reply:
In this sense, I can re-write https://github.com/vitessio/vitess/pull/11103/files#diff-32dbe2effb6b8b7e1151736ee190783bb236feb90ad7c04fe8ec6eec3ebea30fR371 like:

```diff
 func (uvs *uvstreamer) init() error {
-       if uvs.startPos != "" {
+       if uvs.startPos != "" && len(uvs.inTablePKs) > 0 {
+               uvs.pickingUpInputOffset = true
                if err := uvs.setStreamStartPosition(); err != nil {
                        return err
                }
-       } else if uvs.startPos == "" || len(uvs.inTablePKs) > 0 {
                if err := uvs.buildTablePlan(); err != nil {
                        return err
                }
+       } else {
+               if uvs.startPos != "" {
+                       if err := uvs.setStreamStartPosition(); err != nil {
+                               return err
+                       }
+               } else if uvs.startPos == "" || len(uvs.inTablePKs) > 0 {
+                       if err := uvs.buildTablePlan(); err != nil {
+                               return err
+                       }
+               }
        }
```

Although the existing condition in `init()` can remain intact for the safest bet, this code is more redundant.